Update natspec in SaferSafes

alcueca opened the pull request:
No code changes, but auditors might appreciate it.

alcueca committed:
Added a flowchart to the liveness module

alcueca committed:
Added a flowchart to the liveness module

alcueca committed:
Added version compatibility notice

alcueca committed:
Added version compatibility notice

Ethnical reviewed (approved):
LGTM! 
NIT: I wonder if there is another place that we can inform aswell that the only version that is working here is `v1.4.1` maybe in a README or somewhere else

Ethnical commented on packages/contracts-bedrock/src/safe/LivenessModule2.sol:
```diff
@@ -12,9 +12,40 @@ import { GuardManager } from "safe-contracts/base/GuardManager.sol";
+///      +----------------------+
+///      | Start (no challenge) |<---------------------------+
+///      +----------------------+                            |
+///       |                                                  | respond() by Safe
```
NIT: I am not sure to understand why the 
<img width="1167" height="751" alt="image" src="https://github.com/user-attachments/assets/739f57bf-9946-42fd-b91c-1dee0755d916" />
Is connect to the last block here? 
If this match, no fallback ownership should happen right? maybe we can details this more or I might get this confusing.

alcueca replied:
Well, the diagram is like that, because that is the behaviour. You described it correctly.

Anytime after a challenge starts, we can go back to the initial state with any of those three calls.

After the `livenessResponsePeriod` has passed, we can still go back using any of those three calls, or we can change the ownership.

alcueca replied:
Note it doesn't connect to the last block (ownership transferred), but to the one immediately before (ready to transfer ownership).

alcueca replied:
I see now what you mean, I made a state machine diagram (where you start at the Init state, and you are always within the state machine).

You expected a decision tree of what can happen after a challenge call, where you don't go back to the beginning.

Tbh, I prefer to use state machines to describe smart contracts so if you don't mind I'll keep the diagram as it is.

State machine diagrams are very useful in visualizing the workings in smart contracts, in particular forbidden or unexpected transitions. For example, my diagram shows that after the ownership has changed, we don't go back to the start state. Auditors will check that that is the case.

mds1 replied:
I think it's ok to keep the state machine (agreed they are useful), but we should modify it or comment it more if people are having trouble understanding what they are looking at. I suspect a comment that says something like "below is state machine, from "Start" you can follow the arrows to understanding the challenge flows" would clear it up

mds1 commented on packages/contracts-bedrock/src/safe/LivenessModule2.sol:
Needs a patch semver bump due to comment changes

alcueca replied:
Fixed

alcueca committed:
patch version bump

alcueca committed:
Title for the state diagram

alcueca committed:
Title for the state diagram

alcueca committed:
Trimmed comments to 100 characters

alcueca committed:
Trimmed comments to 100 characters

alcueca commented on packages/contracts-bedrock/src/safe/SaferSafes.sol:
```diff
@@ -47,15 +52,16 @@ contract SaferSafes is LivenessModule2, TimelockGuard, ISemver {
 
-        // If the liveness response period is 0, then the liveness module is enabled but not configured.
+        // If the liveness response period is 0, then the liveness module is enabled but not
+        /// configured.
```
That line is 24 characters.

alcueca committed:
Merge branch 'develop' into acc/liveness-module-flowchart

alcueca committed:
One extra slash slashed

alcueca committed:
One extra slash slashed

alcueca committed:
Merge branch 'acc/liveness-module-flowchart' of github.com:ethereum-optimism/optimism into acc/liveness-module-flowchart

alcueca committed:
semver-lock

alcueca committed:
semver-lock

alcueca committed:
semver-lock

aliersh committed:
refactor(prompt): enhance test quality guidance and add commit strategy (#17950)

- add fuzz constraint for bounding value amounts to prevent overflow
- change version testing from length checks to SemverComp.parse() validation
- add version testing example showing wrong vs right approaches
- add commit strategy guidance in methodology phases
- add commit strategy section in PR submission guidelines

alcueca committed:
Merge branch 'acc/liveness-module-flowchart' of github.com:ethereum-optimism/optimism into acc/liveness-module-flowchart

alcueca committed:
Remove the nonce-independent comment

alcueca committed:
We can't do state changes in `checkTransaction`

alcueca committed:
Revert "We can't do state changes in `checkTransaction`"

This reverts commit ae5f394979bcc5631d263ad414c0582564100121.

alcueca commented on packages/contracts-bedrock/src/safe/TimelockGuard.sol:
```diff
@@ -364,10 +374,11 @@ abstract contract TimelockGuard is BaseGuard {
+    ///      been executed. Although making state changes here is a violation of the
+    ///      Checks-Effects-Interactions pattern, it safe to do in this case because we trust that
+    ///      the Safe does not enable arbitrary calls without proper authorization checks. Also, we
+    ///      can't do these state changes in `chackTransaction` because it is a view function.
```
I'll remove this comment completely in a separate PR
